### PR TITLE
restore reposition logic and use onPositioned callback instead

### PR DESCRIPTION
**Pull request by caburj**
After https://github.com/odoo/odoo/commit/3a798039d6f200f8e28448ddb6a2d3c46654a203, the emoji in the chat box is not
properly position. This is because of the change in logic in the application
of the calculated position to take into account the tour pointer.

This commit restores the logic of reposition and then made the tour pointer to use
the onPositioned option to properly reposition its element.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/position_hook.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { localization } from "@web/core/l10n/localization";
  *  margin?: number;
  *  position?: Direction | Position;
  *  onPositioned?: PositionEventHandler;
+ *  customApplyPosition: (popper: HTMLElement, position: object) => void;
```
Comment (brboi) [inline addons/web/static/src/core/position_hook.js:18]: Do not add another option ;-)
Please use `onPositioned`

**Review by brboi — CHANGES_REQUESTED** [review]
Could you also add tests for this fix and also for the changes you introduced in #107618 

Thanks ;-)

File: addons/web/static/src/core/position_hook.js (hunk 1 above)
Comment (caburj) [reply addons/web/static/src/core/position_hook.js:18]: Okay, I'll try that.

File: addons/web/static/src/core/position_hook.js [hunk 2]
```diff
@@ -194,12 +194,11 @@ function getBestPosition(reference, popper, { container, margin, position }) {
...
 /**
- * @param {Options | (() => Options)} options
+ * @param {Options} options
  * @returns {Options}
  */
 function getOptions(options) {
-    const optionsValue = typeof options === "function" ? options() : options;
-    return { ...DEFAULTS, container: document.documentElement, ...optionsValue };
+    return { ...DEFAULTS, container: document.documentElement, ...options };
 }
```
[the remark this answers is not in the document]
Comment (caburj) [reply addons/web/static/src/core/position_hook.js]: Thank you. You're completely right.

[the remark this answers is not in the document]
Comment (brboi) [reply addons/web/static/src/core/position_hook.js]: Thank you for taking care of this fix ;-)
Let me know when the branch is ready 👍🏽

[the remark this answers is not in the document]
Comment (caburj) [reply addons/web/static/src/core/position_hook.js]: It's ready.

**Comment by brboi** [comment]
robodoo r+

**Review by brboi — APPROVED** [review]

**Pull request closed without merging**